app/config/config.js

```diff
@@ -0,0 +1,57 @@
+"use strict";
+
+const localEnv = _loadLocalEnv(),
+    env = process.env.NODE_ENV || 'development';
+
+
+/**
+ * @property {string} HOST
+ * @property {string} PORT
+ * @property {string} MONGODB_URL
+ */
+const config = {
+
+    development: {
+        HOST: localEnv.HOST || '188.166.63.248',
+        PORT: localEnv.PORT || '9030',
+
+        MONGODB_URL: localEnv.MONGODB_URL || 'mongodb://localhost:27017/differ-canvas'
+    },
+
+    test: {
+        HOST: '127.0.0.1',
+        PORT: '3000',
+
+        MONGODB_URL: 'mongodb://localhost:27017/differ-canvas_test_integration'
+    },
+
+    production: {
+        HOST: '127.0.0.1',
+        PORT: '9010',
+
+        MONGODB_URL: 'mongodb://localhost:27017/differ-canvas'
+    }
+};
+
+
+module.exports = config[env];
+
+
+/**
+ * @type function
```

Comment:
See your point, it's definitely valid.

Although we should have as much information here that it makes it easy to spot mistakes, and/or alter the parameter validation etc. Having other middlewares like authentication should definitely be in other places.


```
router.get('/courses',
    isAuthenticated,
    isResourceBelongsTo,
    someOtherMiddleWare,
    validateQuery(Joi.string().max(100).required()),  // if validating the query-string
     validateBody({ // if validating body
           something: joi.string()....
        }), 
        (req, res, next) => {
        CanvasSVC.getCourses(req, res, next);
    });

```

IMO, reason why we should have the validation in this place, because it's in the domain of the API-function and it's easier to spot / alter mistakes or add additional validation without stepping out of context.

So you want to add an API-endpoint. Then these are the steps you would go through:
1. specify the path of the endpoint
2. add generic middleware for the request (isAuthenticated etc.)
3. Add parameter validation. The kind of parameters you want to validate shouldn't be located somewhere else IMO. We're not gonna reuse that parameter validation anyway, so it's only about keeping the code DRY I guess? If we're not gonna reuse it, and it's not a generic middleware, a helper is much better IMO
4. user whatever other service you need to use in the function.